Deduplicator BDD test failure

Due to dependencies being updated, one of our BDDs is now failing. Extracts from the log:

```ShellSession
$ make test
python setup.py test
running pytest
running egg_info
writing requirements to ebu_tt_live.egg-info/requires.txt
writing ebu_tt_live.egg-info/PKG-INFO
writing top-level names to ebu_tt_live.egg-info/top_level.txt
writing dependency_links to ebu_tt_live.egg-info/dependency_links.txt
writing entry points to ebu_tt_live.egg-info/entry_points.txt
reading manifest file 'ebu_tt_live.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'ebu_tt_live.egg-info/SOURCES.txt'
running build_ext
/Users/nigel/Code/ebu/ebu-tt-live-toolkit/venv/lib/python2.7/site-packages/_pytest/config/__init__.py:316: PytestConfigWarning: pytest-capturelog plugin has been merged into the core, please remove it from your requirements.
  name.replace("_", "-")
================================================================================================================= test session starts ==================================================================================================================
platform darwin -- Python 2.7.10, pytest-4.6.4, py-1.5.3, pluggy-0.12.0
rootdir: /Users/nigel/Code/ebu/ebu-tt-live-toolkit, inifile: pytest.ini, testpaths: testing, ebu_tt_live
plugins: bdd-3.1.1, cov-2.7.1, mock-1.10.4, twisted-1.10, capturelog-0.7
collected 746 items                                                                                                                                                                                                                                    

testing/test_config.py .......                                                                                                                                                                                                                   [  0%]
testing/test_consumer.py .                                                                                                                                                                                                                       [  1%]
testing/test_integration.py .                                                                                                                                                                                                                    [  1%]
testing/bdd/test_3350_value_types.py ..................................................                                                                                                                                                          [  7%]
testing/bdd/test_applied_processing.py .....                                                                                                                                                                                                     [  8%]
testing/bdd/test_body_element_content.py ...                                                                                                                                                                                                     [  8%]
testing/bdd/test_bufferDelayNode.py ....                                                                                                                                                                                                         [  9%]
testing/bdd/test_computed_times.py ............................................                                                                                                                                                                  [ 15%]
testing/bdd/test_deduplicator.py .............F                                                                                                                                                                                                  [ 17%]

[snip]

======================================================================================================================= FAILURES =======================================================================================================================
________________________________________ test_successfully_removes_styles_and_regions_of_more_than_one_document[deduplicator_templates/ReSequenced1_12.xml-5-2-deduplicator_templates/ReSequenced1_13.xml-1-1] _________________________________________
Fixture "template_file_two" called directly. Fixtures are not meant to be called directly,
but are created automatically when test functions request them as parameters.
See https://docs.pytest.org/en/latest/fixture.html for more information about fixtures, and
https://docs.pytest.org/en/latest/deprecations.html#calling-fixtures-directly about how to update your code.
----------------------------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------------------------
[<style ID: SEQ58.defaultStyle1 at 0x10a1f3950>, <style ID: SEQ59.defaultStyle1 at 0x10a1f3b50>, <style ID: SEQ60.defaultStyle1 at 0x10a1f3d10>, <style ID: SEQ61.defaultStyle1 at 0x10a1f3ed0>, <style ID: SEQ62.defaultStyle1 at 0x10a1600d0>, <style ID: SEQ63.defaultStyle1 at 0x10a160290>]

```

At this stage it is unclear if the code itself is not operating correctly, if the error is simply the deprecated (now removed) use of fixtures being called directly, or both, or something else entirely.

--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ setup(
         "pyyaml",
         "service_identity",
         "twisted",
-        "autobahn",
+        "autobahn<18",
         "nltk",
         "sortedcontainers",
         "configman",

--- a/testing/bdd/conftest.py
+++ b/testing/bdd/conftest.py
@@ -25,13 +25,20 @@ def template_file_one(xml_file_1):
     return j2_env.get_template(xml_file_1)
 
 @then('a second xml file <xml_file_2>')
-@pytest.fixture
 def template_file_two(xml_file_2):
     cur_dir = os.path.dirname(os.path.abspath(__file__))
     j2_env = Environment(loader=FileSystemLoader(os.path.join(cur_dir, 'templates')),
                          trim_blocks=True)
     return j2_env.get_template(xml_file_2)
 
+# Calling fixtures directly is deprecated, this solution described at
+# https://docs.pytest.org/en/latest/deprecations.html#calling-fixtures-directly
+# seems to work, creating a named fixture rather than defining the "then"
+# step as a fixture directly.
+@pytest.fixture(name='template_file_two')
+def template_file_two_fixture(xml_file_2):
+    return template_file_two(xml_file_2)
+    
 # NOTE: Some of the code below includes handling of SMPTE time base, which was removed from version 1.0 of the specification.
 
 @given('a sequence <sequence_identifier> with timeBase <time_base>')
@@ -97,7 +104,6 @@ def gen_first_document(test_context, template_dict, template_file_one):
     return document1
 
 @then('the second document is generated')
-@pytest.fixture
 def gen_second_document(test_context, template_dict, template_file_two):
     xml_file_2 = template_file_two.render(template_dict)
     document2 = EBUTT3Document.create_from_xml(xml_file_2)
@@ -105,6 +111,13 @@ def gen_second_document(test_context, template_dict, template_file_two):
     document2.validate()
     return document2
 
+# Calling fixtures directly is deprecated, this solution described at
+# https://docs.pytest.org/en/latest/deprecations.html#calling-fixtures-directly
+# seems to work, creating a named fixture rather than defining the "then"
+# step as a fixture directly.
+@pytest.fixture(name='gen_second_document')
+def gen_second_document_fixture(test_context, template_dict, template_file_two):
+    return gen_second_document(test_context, template_dict, template_file_two)
 
 @then('EBUTTD document is valid')
 def then_ebuttd_document_valid(test_context):

